File: frontend/src/lib/api/public.ts
export interface PublicProjectImage {
  id: string;
  image_url: string;
  caption: string | null;
}

export interface PublicProject {
  id: string;
  title: string;
  description: string | null;
  cover_image_url: string | null;
  markdown_content: string | null;
  github_url: string | null;
  live_url: string | null;
  images: PublicProjectImage[];
}

export interface PublicExperience {
  id: string;
  company: string;
  position: string;
  description: string | null;
  start_date: string;
  end_date: string | null;
  is_current: boolean;
  location: string | null;
}

export interface PublicEducation {
  id: string;
  institution: string;
  degree: string;
  field: string | null;
  description: string | null;
  start_date: string;
  end_date: string | null;
  is_current: boolean;
}

export interface PublicSkill {
  id: string;
  name: string;
  category: string | null;
  level: string | null;
  is_transversal: boolean;
}

export interface PublicCertification {
  id: string;
  name: string;
  issuer: string;
  issue_date: string | null;
  credential_url: string | null;
}

export interface PublicFacetResponse {
  slug: string;
  title: string | null;
  bio: string | null;
  meta_title: string | null;
  meta_description: string | null;
  pdf_template: string;
  full_name: string;
  email: string;
  phone: string | null;
  photo_url: string | null;
  website: string | null;
  linkedin_url: string | null;
  github_url: string | null;
  experiences: PublicExperience[];
  educations: PublicEducation[];
  skills: PublicSkill[];
  certifications: PublicCertification[];
  projects: PublicProject[];
}

const BASE_URL = process.env.NEXT_PUBLIC_API_URL ?? "http://localhost:8000";

export const publicApi = {
  getFacet: (slug: string) =>
    fetch(`${BASE_URL}/${slug}`).then((res) => {
      if (!res.ok) throw new Error("Not found");
      return res.json() as Promise<PublicFacetResponse>;
    }),
};

File: backend/src/app/schemas/public.py
import uuid
from datetime import date

from pydantic import BaseModel


class PublicProjectImage(BaseModel):
    id: uuid.UUID
    image_url: str
    caption: str | None

    model_config = {"from_attributes": True}


class PublicProject(BaseModel):
    id: uuid.UUID
    title: str
    description: str | None
    cover_image_url: str | None
    markdown_content: str | None
    github_url: str | None
    live_url: str | None
    images: list[PublicProjectImage] = []

    model_config = {"from_attributes": True}


class PublicExperience(BaseModel):
    id: uuid.UUID
    company: str
    position: str
    description: str | None
    start_date: date
    end_date: date | None
    is_current: bool
    location: str | None

    model_config = {"from_attributes": True}


class PublicEducation(BaseModel):
    id: uuid.UUID
    institution: str
    degree: str
    field: str | None
    description: str | None
    start_date: date
    end_date: date | None
    is_current: bool

    model_config = {"from_attributes": True}


class PublicSkill(BaseModel):
    id: uuid.UUID
    name: str
    category: str | None
    level: str | None
    is_transversal: bool

    model_config = {"from_attributes": True}


class PublicCertification(BaseModel):
    id: uuid.UUID
    name: str
    issuer: str
    issue_date: date | None
    credential_url: str | None

    model_config = {"from_attributes": True}


class PublicFacetResponse(BaseModel):
    slug: str
    title: str | None
    bio: str | None
    meta_title: str | None
    meta_description: str | None
    pdf_template: str

    full_name: str
    email: str
    phone: str | None
    photo_url: str | None
    website: str | None
    linkedin_url: str | None
    github_url: str | None

    experiences: list[PublicExperience] = []
    educations: list[PublicEducation] = []
    skills: list[PublicSkill] = []
    certifications: list[PublicCertification] = []
    projects: list[PublicProject] = []
